src/components/modal.js

```diff
@@ -1,9 +1,47 @@
 import "../polyfills/closest.polyfill";
 import "./w3-dialog";
+import { GetFirstElementOrDefault } from "../utilities/dom.utils";
 
 const cssClasses = {
+  active: "active",
   modalCloseButton: "dg_modal__close-button",
-  modalOpenButton: "dg_modal__open-button"
+  modalOpenButton: "dg_modal__open-button",
+  modalOverlay: "dialog-backdrop",
+  /** overlay is transparent gray section around the modal */
+  canCloseOnOverlayClick: "can-dismiss"
+};
+
+const selectors = {
+  activeDismissibleModal: ".dg_modal[data-dismissible]"
+};
+
+/**
+ * Handle the modal overlay click.
+ * The modal markup will determine whether or not the modal can close on overlay click
+ * based on whether it has the data-dismissible data attribute.
+ * @param {*} clickEvent
+ */
```

Comment:
@martypowell teaching moment....so what exactly is @param {*} doing?

Reply:
@tmccoy529 good question, basically this means that param can be anything, but it shouldn't be.

I could try and explain this but its easier to just point you to the jsdoc, [docs](https://devdocs.io/jsdoc/tags-param)

The reason i started using this format is because if you write your method in vscode, even if you just stub it out

Typing `/**` will populate a little option to add the docs for you.

![image](https://user-images.githubusercontent.com/1933400/67592471-d0ca0400-f72d-11e9-9bf9-d7c7ff3bb73f.png)

All you have to do is to hit enter. I would like to be better about actually documenting the param types, so in this case, ill try and find what the event would be considered.

Reply:
@tmccoy529 this is the best thing i could find, so I will update based on this.

https://stackoverflow.com/a/25045127/1143670

Reply:
updated in the latest change.

Reply:
Okay so this is for documentation kind of like an XML comment in .net? I couldn't see where it had any barring on the functionality so I wasn't sure exactly what it was doing. Thanks for the explanation.

Reply:
yep, not required, but its nice to have, and pretty easy to generate. Works nicely with vscode.

Reply:
@tmccoy529 Thanks for asking that question and thanks @martypowell for providing documentation and an explanation. This is helpful to me as someone who is learning Javascript.